core/src/main/scala/scala/pickling/internal/package.scala

```diff
@@ -24,10 +23,18 @@ package object internal {
 
     }
   }
-  private[this] var currentRuntimeVar = new AtomicReference[spi.PicklingRuntime](initDefaultRuntime)
-  def currentRuntime: spi.PicklingRuntime = currentRuntimeVar.get
-  // Here we inject a new runtime for usage.
-  def replaceRuntime(r: spi.PicklingRuntime): Unit = {
+  private[this] var currentRuntimeVar = new AtomicReference[PicklingRuntime](initDefaultRuntime)
+  def currentRuntime: PicklingRuntime = currentRuntimeVar.get
+
+  /** Replace the old [[PicklingRuntime]] keeping its state.
+    *
+    * Note that we don't do anything with the [[RefRegistry]] because in
+    * future versions we are going to change how cyclic references work.
+    */
+  def replaceRuntime(r: PicklingRuntime): Unit = {
+    if(r.picklers.isLookupEnabled) {
+      currentRuntime.picklers.dumpStateTo(r.picklers)
```

Comment:
A quick note:  This is not thread safe.  It may be ok to assume that we'll only ever swap runtimes on one thread, but if the issue is somehow the runtime is being created on one thread and replaced on another, this won't work.  

I think in the event there are multiple instantiation threads and this function hasn't completed first, we're already screwed, so it's probably ok as is, we just need to *heavily* document how to safely override the runtime and instantiate picklers.


E.g. the documentation for how to repalce runtime right now shows how to create a new object to contain your entire "protocol".  That's part 1 to making sure this is safe.  Part #2 is to make sure you pre-allocate that object in your "main" thread before you spawn sub-threads. 

I had originally punted a bit on this problem, but given we're adding this functionality (which does need to be there), we should really document this problem and possible user solutions.